In [3]:
# Importing the libraries

import os
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [6]:
# Peek into the data by creating pandas dataframe

star_dataframe = pd.read_csv('https://drive.google.com/uc?id=1BQVc6MHjQFtDC9iP1isT_K4ojVe_Oil-')
star_dataframe.sample(10) # Random 10 samples from star_dataframe
# star_dataframe.head() # First 5 samples from start from star_dataframe. We can also mention any number and pass as argument to head() for that many records from start.
# star_dataframe.tail() # Last 5 samples from start from star_dataframe.

Temperature (K)  Luminosity(L/Lo)  Radius(R/Ro)  Absolute magnitude(Mv)  \
37              6380           1.35000         0.980                    2.93   
236            30839      834042.00000      1194.000                  -10.63   
96              5112           0.63000         0.876                    4.68   
58              3752      209000.00000       955.000                  -11.24   
73              3150           0.00880         0.350                   11.94   
72              3304           0.00850         0.180                   13.20   
171             3500      138000.00000      1420.000                   -8.18   
64              2935           0.00014         0.116                   18.89   
230            24145      382993.00000      1494.000                   -8.84   
49             33750      220000.00000        26.000                   -6.10   

     Star type    Star color Spectral Class  
37           3  Yellow-White              F  
236          5          Blue              O  
96           3           Red              K  
58           5           Red              M  
73           1           Red              M  
72           1           Red              M  
171          5           Red              M  
64           0           Red              M  
230          5    Blue-White              B  
49           4          Blue              B

In [8]:
# Check general information about the dataframe.

star_dataframe.info() # Info about different columns and star_dataframe.

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 240 entries, 0 to 239
Data columns (total 7 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Temperature (K)         240 non-null    int64  
 1   Luminosity(L/Lo)        240 non-null    float64
 2   Radius(R/Ro)            240 non-null    float64
 3   Absolute magnitude(Mv)  240 non-null    float64
 4   Star type               240 non-null    int64  
 5   Star color              240 non-null    object 
 6   Spectral Class          240 non-null    object 
dtypes: float64(3), int64(2), object(2)
memory usage: 13.2+ KB


### Observations
**1)** Dataset consists of 240 rows, 6 feature and 1 target columns,
- *Absolute Temperature (in K)*    
- *Relative Luminosity (L/Lo)*
- *Relative Radius (R/Ro)*
- *Absolute Magnitude (Mv)*
- *Star Color*
- *Spectral Class*
- *Star Type (Target classes)* (Target class in reference for Machine Learning Models)
    
Here,
- *Lo = 3.828 x 10^26 Watts (Avg Luminosity of Sun)*
- *Ro = 6.9551 x 10^8 m (Avg Radius of Sun)*
    
**2)** Two categorical features (object type) which will need some kind of encoding,
- *Star Color*
- *Spectral Class*

**3)** It consists of some different features of stars. Information on the star type is given below,       
- **0** → Brown Dwarf       
- **1** → Red Dwarf       
- **2** → White Dwarf      
- **3** → Main Sequence    
- **4** → Supergiants      
- **5** → Hypergiants      

### Create a directory to save different Visualizations

In [9]:
# Create a directory to save the visualizations
folder_name = 'star_plots'                     # directory name
os.makedirs(folder_name, exist_ok=True)        # Create directory if not exisiting
base_dir = f'/content/{folder_name}/'          # Store the path of this directory as base_dir to use it further

### Visualize the count of stars for each type of a star

Matplotlib